clippy_lints/src/lib.rs

```diff
@@ -571,16 +616,7 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
     store.register_late_pass(|| Box::new(non_octal_unix_permissions::NonOctalUnixPermissions));
     store.register_early_pass(|| Box::new(unnecessary_self_imports::UnnecessarySelfImports));
 
-    let msrv = conf.msrv.as_ref().and_then(|s| {
-        parse_msrv(s, None, None).or_else(|| {
-            sess.err(&format!(
-                "error reading Clippy's configuration file. `{}` is not a valid Rust version",
-                s
-            ));
-            None
-        })
-    });
-
+    let msrv = conf.msrv.as_ref().and_then(|s| parse_msrv(s, None, None));
```

Comment:
Why do you have to re-parse it and move the original parsing code to the `read_conf` file.

Couldn't you do something like:

```rust
let mut msrv = <original-parsing-code>;
if let Some(cargo_msrv) = parse_cargo_msrv() {
    if msrv.map_or(false, |msrv| msrv != cargo_msrv) {
        // emit warning
    } else {
        // they are either identical => doesn't matter which
        // or `msrv == None` => use cargo_msrv
        msrv = Some(cargo_msrv)
    }
}
```

Reply:
I think I get your intention. However, I don't really understand why it is necessary to update the `conf.msrv` value. This value is unused after the `let msrv = <conf.msrv parsing code>;` statement AFAIK.

Reply:
I checked that, you're right. It's only used once, so I reworked it to your suggestion.